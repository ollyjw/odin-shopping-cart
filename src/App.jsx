import { useState } from 'react'
import { Outlet } from "react-router-dom"
// import PropTypes from 'prop-types';
import './App.css'
import Navbar from './components/Navbar';

const App = () => {
  const [heading, setHeading] = useState("Odin Project - Shopping Cart App");

  const clickHandler = () => {
    setHeading("Olly's Shopping Cart App");
  };

  return (
    <div>
      <Navbar />
      <Outlet />
      <h1>{heading}</h1>
      <button type="button" onClick={clickHandler}>
        Click Me
      </button>
    </div>
  );
};


export default App
