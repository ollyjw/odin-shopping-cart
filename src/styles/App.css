#root {
  max-width: 1280px;
  width: 100%;
  margin: 0 auto;
  /* padding: 2rem; */
  text-align: center;
}

.app {
  min-height: 100vh;
  padding: 2rem;
}

ul {
  list-style-type: none;
}

h3 {
  margin-top: 0;
}

.lg-screen {
  display: none;
}

/* tablet and above screen sizes */
@media (min-width:768px) {
  .lg-screen {
    display: inline-block;
  }
}