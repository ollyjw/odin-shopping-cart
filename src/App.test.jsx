import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";
import { MemoryRouter } from "react-router-dom";

describe("App component", () => {
  it("renders Odin Project - Shopping Cart App", () => {
    // since screen does not have the container property, we'll destructure render to obtain a container for this test
    const { container } = render(<MemoryRouter><App /></MemoryRouter>);
    expect(container).toMatchSnapshot();
  });

  it("renders Olly's Shopping Cart App after button click", async () => {
    const user = userEvent.setup();

    render(<MemoryRouter><App /></MemoryRouter>);
    const button = screen.getByRole("button", { name: "Click Me" });

    await user.click(button);

    expect(screen.getByRole("heading").textContent).toMatch(/Olly's Shopping Cart App/i);
  });
});
