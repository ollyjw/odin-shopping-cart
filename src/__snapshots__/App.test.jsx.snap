// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders Odin Project - Shopping Cart App 1`] = `
<div>
  <div>
    <nav>
      <ul>
        <li>
          <a
            href="/shop"
          >
            Shop
          </a>
        </li>
      </ul>
    </nav>
    <button
      type="button"
    >
      Click Me
    </button>
    <h1>
      Odin Project - Shopping Cart App
    </h1>
  </div>
</div>
`;
